### PR TITLE
[0.10.7] Report Issue Button and Dialog

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
## Features
- Report Issue Dialog
- Report Issue Button
  Resolves FRO-145 [(link)](https://linear.app/skiso/issue/FRO-145/contact-support-as-a-component)
  - Seen in Landing, 404, Client-side Error Fallback, and Name Change Dialog
- GGC link is now always to the right account — thanks to @pixelpxed for method discovery

## Fixes
- `/learn/electives` is accessible by Teachers, and `teach/electives` is accessible by Students
  Fixes FRO-155
- Student Attendance Summary thinks the date is always January 9
  Fixes FRO-170
- Birthday is invalid in vCard
  Fixes FRO-171
- Some Contacts’ label is “undefined” in vCard
  Fixes FRO-172
- FedCM UI blocks the footer with no way to scroll to see it
- Room is now hidden in Single Subject Details if no data is available

## Preview
Preview available at https://pr3.mysk.school/.

### CODE REVIEW

**Review by Sun-52 — APPROVED** [review]
LGTM

**Review by smartwhatt — APPROVED** [review]
LGTM!

File: components/account/about/NameChangeDialog.tsx
Comment (pixelpxed) [inline components/account/about/NameChangeDialog.tsx:1]: Interesting that this file is called "NameChangeDialog.tsx".

File: components/home/ClassroomSubjectCard.tsx [hunk 1]
```diff
@@ -32,6 +33,18 @@ const ClassroomSubjectCard: StylableFC<{
   const locale = useLocale();
   const { t } = useTranslation("schedule", { keyPrefix: "subjectList.card" });
 
+  const mysk = useMySKClient();
+
+  /** If applicable, ensure the URL is to the right account. */
+  const url = subject.ggc_link
+    ? mysk.user?.email
+      ? `https://accounts.google.com/AccountChooser?${new URLSearchParams({
```
Comment (pixelpxed) [inline components/home/ClassroomSubjectCard.tsx:41]: Having a normal Google Classroom link as a fallback is a good catch if there's an error.

File: utils/helpers/contact/useGetVCard.ts [hunk 2]
```diff
@@ -64,7 +63,7 @@ export default function useGetVCard() {
 
           // VCard metadata
           `KIND:individual`,
-          `REV:${new Date().toISOString()}`,
+          `REV:${new Date().toISOString().replaceAll(/[-:.]/g, "")}`,
```
Comment (pixelpxed) [inline utils/helpers/contact/useGetVCard.ts:66]: Good catch!

**Review by pixelpxed — APPROVED** [review]
LGTM!

**Pull request merged**